packages/dd-trace/src/plugins/util/git.js

```diff
@@ -36,8 +36,24 @@ function isShallowRepository () {
 }
 
 function unshallowRepository () {
-  sanitizedExec('git', ['config', 'remote.origin.partialclonefilter', '"blob:none"'])
-  sanitizedExec('git', ['fetch', '--shallow-since="1 month ago"', '--update-shallow', '--refetch'])
+  const gitVersionString = sanitizedExec('git', ['version'])
+  const gitVersionSplit = gitVersionString.split(' ')[2].split('.')
+  const gitVersionMain = parseInt(gitVersionSplit[0])
+  const gitVersionSecondary = parseInt(gitVersionSplit[1])
+  if (gitVersionMain <= 2 && gitVersionSecondary < 27) {
+    return
+  }
+  const defaultRemoteName = sanitizedExec('git', ['config', '--default', 'origin', '--get', 'clone.defaultRemoteName'])
+  const revParseHead = sanitizedExec('git', ['rev-parse', 'HEAD'])
+  sanitizedExec('git', [
+    'fetch',
+    '--shallow-since="1 month ago"',
+    '--update-shallow',
+    '--filter="blob:none"',
```

Comment:
this is also important, otherwise the command does not work
```suggestion
    '--filter=blob:none',
```